libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/RoadObjectDistanceInfo.kt

```diff
@@ -0,0 +1,62 @@
+package com.mapbox.navigation.base.trip.model.roadobject
+
+/**
+ * RoadObjectDistanceInfo contains information about distance to the road object of a concrete
+ * flavor/shape (gantry, polygon, line, point etc.)
+ *
+ * Available types are:
+ * - [RoadObjectDistanceInfoType.GANTRY]
+ * - [RoadObjectDistanceInfoType.LINE]
+ * - [RoadObjectDistanceInfoType.POINT]
+ * - [RoadObjectDistanceInfoType.POLYGON]
+ * - [RoadObjectDistanceInfoType.SUB_GRAPH]
+ *
+ * @param roadObjectId id of the road object
+ * @param roadObjectType type of the road object
+ * @param distanceInfoType type of the distance info object
+ */
+abstract class RoadObjectDistanceInfo(
```

Comment:
The constructor here needs to be internal.

Reply:
Or an internal constructor and a factory in the internal package which I think might be the best option :)

Reply:
done